### PR TITLE
WE-601 fix CWE-171 log forge warning

### PR DESCRIPTION
**Pull request by steinsiv**
## Fixes
This pull request fixes `WE-601`

## Description
[CWE-117](https://cwe.mitre.org/data/definitions/117.html) Sanitation of Header values. 
Strip data not matching the expected input (JWT token, in this case)

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Services/CredentialsService.cs [hunk 1]
```diff
@@ -146,6 +147,7 @@ public async Task<(string username, string witsmlUsername)> GetUsernames()
         {
             StringValues authorizationHeader = _httpContextAccessor.HttpContext.Request.Headers["Authorization"];
             string bearerToken = authorizationHeader.Count > 0 ? authorizationHeader.ToString().Split()[1] : null;
+            bearerToken = Regex.Match(bearerToken, "[a-zA-Z0-9-_.]+").Value;
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/CredentialsService.cs:149]: Will crash on null

Comment (steinsiv) [reply Src/WitsmlExplorer.Api/Services/CredentialsService.cs:149]: Good point!

File: Src/WitsmlExplorer.Api/Services/CredentialsService.cs [hunk 2]
```diff
@@ -145,9 +146,10 @@ public bool ValidEncryptedBasicCredentials(string headerName)
         public async Task<(string username, string witsmlUsername)> GetUsernames()
         {
             StringValues authorizationHeader = _httpContextAccessor.HttpContext.Request.Headers["Authorization"];
-            string bearerToken = authorizationHeader.Count > 0 ? authorizationHeader.ToString().Split()[1] : null;
+            string bearerToken = authorizationHeader.Count > 0 ? authorizationHeader.ToString().Split()[1] : string.Empty;
+            bearerToken = Regex.Match(bearerToken, "[a-zA-Z0-9-_.]+").Value;
             ServerCredentials creds = await GetCredentials(WitsmlClientProvider.WitsmlTargetServerHeader, bearerToken);
-            if (bearerToken != null)
+            if (string.IsNullOrEmpty(bearerToken))
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/CredentialsService.cs]: Logic was inverted from "bearerToken is not null" to "bearerToken is null" - missing exclamation mark

**Review by janburak — DISMISSED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [ ] Verify that PR resolves issue
- [X] Reviewed the code

**Review by janburak — DISMISSED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Review by janburak — APPROVED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Pull request merged**